frontend/js/src/user/stats/components/UserArtistActivity.tsx

```diff
@@ -166,28 +201,39 @@ export default function UserArtistActivity(props: UserArtistActivityProps) {
                     tickSize: 5,
                     tickPadding: 5,
                     tickRotation: -45,
-                    renderTick: (tick) => (
-                      <g transform={`translate(${tick.x},${tick.y})`}>
-                        {tick.value
-                          .split("\n")
-                          .map((line: string, i: number) => (
-                            <text
-                              key={line}
-                              x={0}
-                              y={10 + i * 15}
-                              textAnchor="end"
-                              dominantBaseline="middle"
-                              style={{
-                                fontSize: 10,
-                                fill: "#000",
-                                transform: `rotate(-45deg)`,
-                              }}
-                            >
-                              {line}
-                            </text>
-                          ))}
-                      </g>
-                    ),
+                    renderTick: (tick) => {
+                      const artistMbid =
+                        rawData?.result?.[tick.tickIndex]?.artist_mbid || "";
+                      const artistName =
+                        rawData?.result?.[tick.tickIndex]?.name || "";
+                      const lines = tick.value.split("\n");
+                      const linkTo = artistMbid
+                        ? `/artist/${artistMbid}`
+                        : `/search?search_term=${encodeURIComponent(
+                            artistName
+                          )}&search_type=artist`;
+                      return (
+                        <g transform={`translate(${tick.x},${tick.y})`}>
+                          <foreignObject
+                            x="-90"
+                            y="0"
+                            width="90"
+                            height="30"
+                            style={{
+                              overflow: "visible",
+                              transform: "translate(-5px, 5px) rotate(-45deg)",
+                              textAlign: "right",
+                              fontSize: "11px",
+                              lineHeight: "1.25em",
+                            }}
+                          >
+                            <Link to={linkTo} className="ellipsis-3-lines">
+                              {lines.join("\n")}
```

Comment:
We are now splitting on newlines and reconstituting it afterwards for not reason :)
You'll need to remove the `const lines = ...` line above
```suggestion
                              {tick.value}
```